### PR TITLE
Add simple datetime module [WIP]

### LINKED ISSUE #67: Add time util module
something like [moment](https://github.com/moment/moment) or [date-fns](https://github.com/date-fns/date-fns)

Let it be an official library.

Reduce developer focus, focus on their code rather than due to which library to choose

is there anyone think so

### PR DESCRIPTION
**Pull request by sh7dm**
Add simple datetime module. **:construction: WIP :construction:**

Resolves #67.

### CODE REVIEW

File: datetime/index.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+export const parseDate = (date, format: string) => {
```
Comment (ry) [inline datetime/index.ts]: rename this file to mod.ts

File: datetime/index.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  return {
+    year: Number(y),
+    month: Number(m),
+    day: Number(d)
+  }
+}
+
+export const parseDateTime = (datetime, format: string) => {
```
Comment (ry) [inline datetime/index.ts]: `export function parseDateTime(datetimeStr: string , format: string): DateTime`

File: datetime/index.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    year: Number(y),
+    month: Number(m),
+    day: Number(d),
+    hours: Number(ho),
+    minutes: Number(mi)
+  }
+}
+
+export const currentDayOfYear = () => Math.floor(
+  (new Date().getMilliseconds() - new Date(new Date().getFullYear(), 0, 0).getMilliseconds()) / 1000 / 60 / 60 / 24
```
Comment (ry) [inline datetime/index.ts]: Run format.ts

File: datetime/example.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import * as datetime from 'index.ts'
+
+console.log(datetime.currentDayOfYear())
+console.log(datetime.parseDateTime("01-03-2019 16:34", "mm-dd-yyyy hh:mm"))
+console.log(datetime.parseDate("01-03-2019", "mm-dd-yyyy"))
```
Comment (ry) [inline datetime/example.ts]: Rename this to `//datetime/test.ts` and import it in `//test.ts`

File: datetime/index.ts (hunk 1 above)
Comment (sh7dm) [reply datetime/index.ts]: @ry rename index to mod?

Comment (ry) [reply datetime/index.ts]: yes

**Comment by sh7dm** [comment]
@ry I've made all the changes

File: test.ts [hunk 5]
```diff
@@ -19,6 +19,7 @@ import "path/relative_test.ts";
 import "path/resolve_test.ts";
 import "path/zero_length_strings_test.ts";
 import "testing/test.ts";
+import "datetime/test.ts";
```
Comment (ry) [inline test.ts]: this import list should be alphabetical

File: datetime/mod.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    [, d, m, y] = datePattern.exec(dateStr);
+  } else if (format === "yyyy-mm-dd") {
+    const datePattern = /^(\d{4})-(\d{2})-(\d{2})$/;
+    [, y, m, d] = datePattern.exec(dateStr);
+  }
+
+  return new Date(Number(y), Number(m) - 1, Number(d));
+};
+
+export const parseDateTime = (datetimeStr: string, format: string): Date => {
```
Comment (ry) [inline datetime/mod.ts]: use `function` syntax not arrows

File: datetime/mod.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
+export const parseDate = (dateStr: string, format: string): Date => {
```
Comment (ry) [inline datetime/mod.ts]: use `function` syntax not arrows

File: datetime/mod.ts (hunk 6 above)
Comment (ry) [reply datetime/mod.ts]: add JS Doc please

File: datetime/mod.ts (hunk 7 above)
Comment (ry) [reply datetime/mod.ts]: add JS Doc please

File: datetime/mod.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    [, ho, mi, d, m, y] = datePattern.exec(datetimeStr);
+  } else if (format === "hh:mm yyyy-mm-dd") {
+    const datePattern = /^(\d{4})-(\d{2})-(\d{2}) (\d{2}):(\d{2})$/;
+    [, ho, mi, y, m, d] = datePattern.exec(datetimeStr);
+  }
+
+  return new Date(Number(y), Number(m) - 1, Number(d), Number(ho), Number(mi));
+};
+
+export const currentDayOfYear = () =>
```
Comment (ry) [inline datetime/mod.ts]: add JS Doc please  and use function syntax

File: datetime/test.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { test, assertEqual } from "../testing/mod.ts";
+import * as datetime from "mod.ts"
+
+test(function parseDateTime() {
+  assertEqual(datetime.parseDateTime("01-03-2019 16:34", "mm-dd-yyyy hh:mm"), new Date(2019, 1, 3, 16, 34));
+});
+test(function parseDate() {
+  assertEqual(datetime.parseDateTime("01-03-2019", "mm-dd-yyyy"), new Date(2019, 1, 3));
+});
```
Comment (ry) [inline datetime/test.ts]: Add test for currentDayOfYear

Comment (sh7dm) [reply datetime/test.ts]: @ry how? it's value depends on current datetime

File: datetime/mod.ts (hunk 8 above)
Comment (sh7dm) [reply datetime/mod.ts]: Ok

File: datetime/mod.ts (hunk 7 above)
Comment (sh7dm) [reply datetime/mod.ts]: Ok

File: datetime/mod.ts (hunk 6 above)
Comment (sh7dm) [reply datetime/mod.ts]: Ok

File: test.ts (hunk 5 above)
Comment (sh7dm) [reply test.ts]: Ok, i'll move it

File: datetime/mod.ts [hunk 10]
```diff
@@ -15,7 +22,14 @@ export const parseDate = (dateStr: string, format: string): Date => {
...
 
-export const parseDateTime = (datetimeStr: string, format: string): Date => {
+/**
+ * Parse date & time from string using format string
+ *
+ * @param {string} dateStr - date & time string
+ * @param {string} format - format string
+ * @return {Date} Parsed date
+ */
+export function parseDateTime (datetimeStr: string, format: string): Date {
```
Comment (ry) [inline datetime/mod.ts]: Run format.ts

Comment (sh7dm) [reply datetime/mod.ts]: OK!

File: datetime/test.ts (hunk 9 above)
Comment (ry) [reply datetime/test.ts]: use Date.now()

Comment (sh7dm) [reply datetime/test.ts]: OK

**Pull request merged**